operator/control-plane/src/market.rs

```diff
@@ -2753,178 +2490,128 @@ mod tests {
             (100, Action::MetadataUpdated, "{\"region\":\"ap-south-1\",\"url\":\"https://example.com/updated-enclave.eif\",\"instance\":\"c6a.large\",\"memory\":4096,\"vcpu\":2}".to_string().abi_encode()),
             (505, Action::Close, [].into()),
-        ].into_iter().map(|x| (x.0, test::get_log(x.1, Bytes::from(x.2), job_num))).collect();
+        ];
 
-        let start_time = Instant::now();
-        // pending stream appended so job stream never ends
-        let job_stream = std::pin::pin!(tokio_stream::iter(job_logs.into_iter())
-            .then(|(moment, log)| async move {
-                let delay = start_time + Duration::from_secs(moment) - Instant::now();
-                sleep(delay).await;
-                log
-            })
-            .chain(tokio_stream::pending()));
-        let mut aws: TestAws = Default::default();
-        let res = market::job_manager_once(
-            job_stream,
-            &mut aws,
-            market::JobId {
-                id: job_num.encode_hex_with_prefix(),
+        let job_manager_params = JobManagerParams {
+            job_id: market::JobId {
+                id: job_id.clone(),
                 operator: "abc".into(),
                 contract: "xyz".into(),
                 chain: "123".into(),
             },
-            &["ap-south-1".into()],
-            300,
-            &test::get_rates(),
-            &test::get_gb_rates(),
-            &Vec::new(),
-            &Vec::new(),
-        )
-        .await;
+            allowed_regions: vec!["ap-south-1".to_owned()],
+            address_whitelist: vec![],
+            address_blacklist: vec![],
+        };
 
-        // job manager should have errored out
-        assert_eq!(res, -2);
-        assert!(aws.outcomes.is_empty());
-        assert!(!aws.instances.contains_key(&job_num.to_string()))
+        let test_results = TestResults {
+            res: -2,
+            outcomes: vec![],
+        };
+
+        run_test(start_time, logs, job_manager_params, test_results).await;
     }
 
     #[tokio::test(start_paused = true)]
     async fn test_metadata_update_event_with_no_updates_after_spin_up() {
-        let _ = market::START.set(Instant::now());
+        let start_time = Instant::now();
+        let job_id = format!("{:064x}", 1);
 
-        let job_num = U256::from(1).into();
-        let job_logs: Vec<(u64, Log)> = vec![
-            (0, Action::Open, ("{\"region\":\"ap-south-1\",\"url\":\"https://example.com/enclave.eif\",\"instance\":\"c6a.xlarge\",\"memory\":4096,\"vcpu\":2}".to_string(),31000000000000u64,31000u64,market::now_timestamp().as_secs()).abi_encode_sequence()),
+        let logs = vec![
+            (0, Action::Open, ("{\"region\":\"ap-south-1\",\"url\":\"https://example.com/enclave.eif\",\"instance\":\"c6a.xlarge\",\"memory\":4096,\"vcpu\":2}".to_string(),31000000000000u64,31000u64,0).abi_encode_sequence()),
             // instance type has also been updated in the metadata. should fail this job.
```

Comment:
remove this comment, there is not change in instance type. Also, in this test, instance should be running instead of failing since there isn't any update, please check.